packages/app-builder/src/routes/ressources+/data+/createLink.tsx

```diff
@@ -104,6 +104,33 @@ export function CreateLink({
   otherTables: TableModel[];
   thisTable: TableModel;
   children: React.ReactNode;
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <Modal.Root open={isOpen} onOpenChange={setIsOpen}>
+      <Modal.Trigger asChild>{children}</Modal.Trigger>
+      <Modal.Content>
+        <CreateLinkContent
+          thisTable={thisTable}
+          otherTables={otherTables}
+          closeModal={() => {
+            setIsOpen(false);
+          }}
+        />
+      </Modal.Content>
+    </Modal.Root>
+  );
+}
+
+function CreateLinkContent({
+  thisTable,
+  otherTables,
+  closeModal,
+}: {
+  otherTables: TableModel[];
+  thisTable: TableModel;
+  closeModal: () => void;
```

Comment:
Do not read this file, mostly a separation between modal and modal content to match existing convention and ease the "auto reset" given by mounting/unmounting modal content comp.